### ISSUE TITLE
Feature request: Some way of highlighting lines of code within a script

### ISSUE THREAD

**Issue by fherreazcue** [carpentries/sandpaper#615]
It is fairly common in lessons in which a script is being developed that the full script is reproduced on every step.
Sometimes, this makes it a bit hard to follow which are the line that changed in the last step.
It would be useful to be able to highlight one or multiple lines of code within a code block.
For example, 
```python
import pandas as pd

file_path="/my/file.csv"

def stats():
  with open(file_path) as f:
    df = pd.read_csv(f)
    df = df.iloc[:, 1:]                 <---- make this line bold, or a different background
    stats = df.describe()
    return stats.to_json()
```
One option is markdown is to use the `diff` language, but it does not work with sandpaper. Also, you loose the syntax highlighting.

Potentially useful as a substitute, if this is not possible, would be to allow turning on code line-numbers, so that they can be pointed out.

**Comment by froggleston** [comment]
Hey @fherreazcue ! As sandpaper is based on basic markdown syntax, I'm not sure this is possible, at least at the moment. It would be good to see if anyone else has any experience!

**Comment by froggleston** [comment]
Similarly, I don't believe pkgdown supports myst syntax as it's a different flavour of markdown. I think the most likely route to getting this implemented would be the larger piece of work to move sandpaper over into quarto: https://github.com/quarto-dev/quarto-cli/discussions/2987

**Comment by brownsarahm** [comment]
I did not realize how much the workbench relied on something not easily extensible (pkgdown) I expected the example of how myst does it (it's open source) to be salient to implementing it for the workbench.

**Comment by froggleston** [comment]
It's not so much that pkgdown isn't extensible, it's more that it provides static site generation functionality that is very beneficial to us (alongside the wide usage of it in the community, and very active development community by experienced core R developers). Granted, it does only support Rmarkdown and knitr based source documents, which is why there is a community push to incorporate proper Quarto support into pkgdown, which would greatly increase the feature set.

Unless I'm mistaken, I'm not sure there's any R tooling available for myst as they are focused on a python ecosystem, which would also mean we'd have to build up a parser from scratch...
